---
title: "RNA-Seq Analysis SRP022028 Revision"
author: "Trent Gutierrez"
date: "2022-09-27"
output: 
  html_document:
    theme: "lumen"
    code_folding: hide
    toc: true
    toc_float: 
      collapsed: false
      smooth_scroll: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


## Introduction

### Research question

This RNA-Seq analysis looks at data from iPSCs derived from patients of 22q11.2 deletion syndrome and how it affects neuronal development and activity pathways. 22q11.2 DS is one of the highest known risk factors for developing schizophrenia (SZ) and research has been limited due to the fact that analyzing developing human brains is inaccessible, that's where Induced Pluripotent Stem Cells (iPSC) come in, they help give researchers a opportunity to grow patient-specific neurons in vitro, and these cells can have the abnormalities needed for the investigation. The data for this analysis will be 8 samples of 14 day old neurons, where 4 are from SZ patients (2 confirmed 22q11.2 DS samples, 2 uncertain patient samples) and 4 controls of normal neurons. 




```{r library loading, message=FALSE, warning=FALSE, include=FALSE}
library(tidyverse)
library(BiocManager)
library(recount)
library(RColorBrewer)
library(pheatmap)
library(DT)
library(DESeq2)
library(fgsea)
library(dplyr)
library(ggplot2)
library(plotly)
library(styler)
library(kableExtra)
library(fgsea)
library(stats)
library(clusterProfiler)
library(enrichplot)
library(pathview)
library(org.Hs.eg.db)
library(devtools)
library(conflicted)
library(stringr)
library(fgsea)
library(data.table)
library(biomaRt)
library(EnhancedVolcano)
```

### Data Input

```{r data input, message=FALSE, warning=FALSE, cache=TRUE, include=FALSE}
url <- download_study("SRP022028") 
load(file.path("SRP022028", 'rse_gene.Rdata'))
```



```{r condition setup, message=FALSE, warning=FALSE, cache=TRUE, include=FALSE}
metadata <- colData(rse_gene)

metadata$conditions <- sapply(metadata$characteristics, function(x){ x[1]})

metadata$genotype <- grepl("22q11", metadata$title) %>%
  ifelse("22q11Del","WT")
```

This is the metadata for our RNA-Seq analysis, the rest of the analysis will use this data to identify what each sample is. 

```{r metadata table, echo=TRUE, message=FALSE, warning=FALSE}
metadata %>% 
  as.data.frame() %>%
  dplyr::select(genotype,conditions) %>%
  kbl(caption = "Metadata for SRP022028") %>%
  kable_styling(bootstrap_options = "striped", full_width = T, html_font = "Cambria")
```

```{r DESeq2 analysis start, message=FALSE, warning=FALSE, cache=TRUE, include=FALSE}
rse_gene$genotype  <- metadata$genotype

rse_gene$condition <- metadata$conditions %>%
  gsub(pattern = "sample group: patients", replacement = "patients") %>%
  gsub(pattern = "sample group: contol", replacement = "control")  

dds <- DESeqDataSet(rse_gene, design = ~condition)
dds <- DESeq(dds)
```



```{r dds result extraction, message=FALSE, warning=FALSE, cache=TRUE, include=FALSE}
result <- results(dds, contrast = c("condition","patients","control"))

result_dataframe <- as.data.frame(result)

fullensemblID <- rownames(result_dataframe) %>%
  str_replace(pattern = "\\..*",replacement = "")
```



## Results

### Exploratory Analysis 

### PCA Plot
```{r PCA plot, echo=TRUE, message=FALSE, warning=FALSE}
vsd_dds <- vst(dds)

plotPCA(vsd_dds, intgroup = c("condition","genotype"))
```




### Statistical Analysis 



```{r signifcant DEGs selection, cache=TRUE, include=FALSE}
sigresults_df <- result %>%
  as.data.frame() %>%
  dplyr::mutate(ID = gsub(pattern = "\\..*", replacement = "", x= rownames(result))) %>%
  dplyr::relocate(ID,baseMean) %>%
  arrange(desc(stat)) %>%
  subset(padj < 0.05) 
```


```{r Gene symbol ID selection, cache=TRUE, include=FALSE}
ensembl <- useEnsembl(biomart = "genes", dataset = "hsapiens_gene_ensembl")

sigresult_gene_id <- getBM(attributes = c("ensembl_gene_id","hgnc_symbol"),
                           filters = "ensembl_gene_id",
                           values = sigresults_df$ID,
                           mart = ensembl) %>%
                     dplyr::rename(ID = ensembl_gene_id, 
                                   Gene_name = hgnc_symbol)

sigresults_df <- left_join(sigresults_df,sigresult_gene_id, by= "ID") %>%
  dplyr::relocate(Gene_name,baseMean) %>%
  dplyr::select(ID,Gene_name,log2FoldChange,stat,pvalue,padj)

# Removes the rows with No gene names, not sure which we would prefer for the data table. If I had to guess I would assume we would want a table with Gene names so the results are more straightforward and we at least know each ensemble ID is unique and doesn't have a chance of associated with another ID.   
sigresults_df_filtered <- sigresults_df %>%
  drop_na(Gene_name) %>%
  dplyr::filter(Gene_name != "")
  
```

### Significant DEGs table

```{r sig DEGs datatable, echo=TRUE, message=FALSE, warning=FALSE}
datatable(sigresults_df_filtered, class = 'cell-border stripe' , 
          caption = "Table 1: Patient vs Control Significant Genes")
```






### Enhanced Volcano plot


```{r enhanced Volcano plot, message=FALSE, warning=FALSE}
EnhancedVolcano(sigresults_df_filtered, 
                lab = sigresults_df_filtered$Gene_name,
                x = 'log2FoldChange',
                y = 'pvalue',
                title ="volcano Slot of Signifcant DEGs",
                xlim = c(-30,30),
                ylim = c(0,20),
                pCutoff = 0.05,
                FCcutoff = 0.5,
                pointSize = 2.0,
                labSize = 3.0 )
```




### Heatmap


```{r heatmap data input, cache=TRUE, include=FALSE}

topover <- dplyr::slice_max(sigresults_df_filtered,n = 10, order_by = stat)
topunder <- dplyr::slice_min(sigresults_df_filtered,n = 10, order_by = stat)
exprdata <- rbind(topover,topunder)

normalized_dds_counts <- counts(dds, normalized=TRUE)
row.names(normalized_dds_counts) <- str_replace(fullensemblID,
                          pattern = "\\..*", 
                          replacement = "")

sig_norm_dds_counts <- normalized_dds_counts[exprdata$ID,]
row.names(sig_norm_dds_counts) <- (exprdata$Gene_name)



heatmeta <- as.data.frame(metadata) %>%
  dplyr::select(conditions,genotype)




vsd <- vst(dds)
sampledist <- dist(t(assay(vsd)))
sampledistmatrix <- as.matrix(sampledist)

rownames(sampledistmatrix) <- paste(vsd$condition, vsd$genotype, sep="-")
colnames(sampledistmatrix) <- NULL

colors <- colorRampPalette(rev(brewer.pal(9, "Blues")) )(255)

cols <- brewer.pal(9, "PuRd")
  
palette <- colorRampPalette(cols)

```



```{r heatmap plot, message=FALSE, warning=FALSE}
heatmap <- pheatmap(sig_norm_dds_counts,
                    color = palette(20),
                    cluster_rows = TRUE,
                    show_rownames = TRUE,
                    annotation = dplyr::select(heatmeta,conditions),
                    scale = "row")
```


NOTE: I think a heatmap like this would be better used at the beginning of the paper, it looks like exploratory data. 
```{r Heatmap sample to sample distances, message=FALSE, warning=FALSE}
distheatmap <- pheatmap(sampledistmatrix,
                        clustering_distance_rows=sampledist,
                        clustering_distance_cols=sampledist,
                        col=colors)
```



```{r Clusterprofiler set up as backup, message=FALSE, warning=FALSE, cache=TRUE, include=FALSE}

#gene_list <- result_dataframe$stat %>%
  #names(gene_list) <- make.names(names(fullensemblID)) %>%
  #as_tibble() %>%
  #dplyr::mutate(ID = fullensemblID) %>%
  #drop_na() %>%
  #dplyr::arrange(desc(value))
  


original_gene_list <- result_dataframe$stat
names(original_gene_list) <- fullensemblID
gene_list <- na.omit(original_gene_list)
gene_list <- sort(gene_list, decreasing = TRUE)

```

```{r Gene mapped, message=FALSE, warning=FALSE, cache=TRUE, include=FALSE}
gse <- gseGO(geneList = gene_list , ont = "ALL",
             keyType = "ENSEMBL",
             pvalueCutoff = 0.05, verbose = TRUE,
             OrgDb= "org.Hs.eg.db", pAdjustMethod = "holm")
```

### GSEA2 Plot for Top 5 Overexpressed and Underexpressed DEGs 

#### Top 5 Overexpressed Pathways


```{r GSEA2 Plot overexpressed, message=FALSE, warning=FALSE}
overGSEAplot <- gse %>%
  arrange(rank) %>%
  gseaplot2(geneSetID = 1:5, title = "Top 5 Overexpressed Pathways")
overGSEAplot
```

#### Top 5 Underexpressed



```{r GSEA2 Plot underexpressed, message=FALSE, warning=FALSE}
underGSEAplot <- gse %>%
  arrange(desc(rank)) %>%
  gseaplot2(geneSetID = 1:5, title = "Top 5 Underexpressed Pathways")
underGSEAplot
```

Discussion: 

Conclusion: 

References: 
