---
title: "RNA-Seq Analysis SRP022028 Revision"
author: "Trent Gutierrez"
date: "2022-09-27"
output: 
  html_document:
    theme: "lumen"
    code_folding: hide
    toc: true
    toc_float: 
      collapsed: false
      smooth_scroll: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


## Introduction

### Research question
This RNA-Seq analysis looks at induced pluripotent stem cells (iPSCs) derived neurons from patients of Schizophrenia and Schizo-affective disorder that also had 22q11.2 deletion syndrome which is one of the most common risk factors for the development of Schizophrenia. The control group is neurons from healthy individuals. The researchers who collected this data were investigating if samples from patients with 22q11.2 DS could serve as a useful genetic pattern for further Schizophrenia research.  

```{r library loading, message=FALSE, warning=FALSE, include=FALSE}
library(tidyverse)
library(BiocManager)
library(recount)
library(RColorBrewer)
library(pheatmap)
library(DT)
library(DESeq2)
library(fgsea)
library(dplyr)
library(ggplot2)
library(plotly)
library(styler)
library(kableExtra)
library(fgsea)
library(stats)
library(clusterProfiler)
library(enrichplot)
library(pathview)
library(org.Hs.eg.db)
library(devtools)
library(conflicted)
library(stringr)
library(fgsea)
library(data.table)
library(biomaRt)
library(EnhancedVolcano)
```

### Data Input

```{r data input, message=FALSE, warning=FALSE, cache=TRUE, include=FALSE}
url <- download_study("SRP022028") 
load(file.path("SRP022028", 'rse_gene.Rdata'))
```



```{r condition setup, message=FALSE, warning=FALSE, cache=TRUE, include=FALSE}
metadata <- colData(rse_gene)

metadata$conditions <- sapply(metadata$characteristics, function(x){ x[1]})

metadata$genotype <- grepl("22q11", metadata$title) %>%
  ifelse("22q11Del","WT")
```

Disclaimer: The original data-set has 18 samples, however during the data acquisition only 8 samples could be downloaded for analysis. 

The metadata shows us which samples are from the patients and controls, and what genotypes the samples also have. 

```{r metadata table, echo=TRUE, message=FALSE, warning=FALSE}
metadata %>% 
  as.data.frame() %>%
  dplyr::select(genotype,conditions) %>%
  kbl(caption = "Table 1: Metadata for SRP022028") %>%
  kable_styling(bootstrap_options = "striped", full_width = T, html_font = "Cambria")
```


```{r DESeq2 analysis start, message=FALSE, warning=FALSE, cache=TRUE, include=FALSE}
rse_gene$genotype  <- metadata$genotype

rse_gene$condition <- metadata$conditions %>%
  gsub(pattern = "sample group: patients", replacement = "patients") %>%
  gsub(pattern = "sample group: contol", replacement = "control")  

dds <- DESeqDataSet(rse_gene, design = ~condition)
dds <- DESeq(dds)
```


```{r dds result extraction, message=FALSE, warning=FALSE, cache=TRUE, include=FALSE}
result <- results(dds, contrast = c("condition","patients","control"))

result_dataframe <- as.data.frame(result)

fullensemblID <- rownames(result_dataframe) %>%
  str_replace(pattern = "\\..*",replacement = "")
```


## Results

### Exploratory Analysis 

```{r sample to sample dist data input, message=FALSE, warning=FALSE, include=FALSE}
sampledist <- vst(dds) %>%
  assay() %>%
  t() %>%
  dist()


sampledistmatrix <- as.matrix(sampledist)

vsd_dds <- vst(dds)

rownames(sampledistmatrix) <- paste(vsd_dds$condition, vsd_dds$genotype, sep="-")
colnames(sampledistmatrix) <- NULL

colsBlue <- colorRampPalette(rev(brewer.pal(9, "Blues")))(255)
```
### Sample-to-Sample dist Heatmap

To begin our analysis we want to view how similar our samples are to each other. In this dist Heatmap our samples cluster in a expected way, 3 controls are close to each other while the 4 patient samples are similar to their group. One interesting observation is that one control-WT sample is very similar to the patient samples. 
```{r Heatmap sample to sample distances, echo=TRUE, message=FALSE, warning=FALSE}
distheatmap <- pheatmap(sampledistmatrix,
                        clustering_distance_rows=sampledist,
                        clustering_distance_cols=sampledist,
                        col=colsBlue)
```


### PCA Plot

This PCA plot helps us continue our sample clustering analysis, here we can see a control-WT close to the patient groups again. And the rest of the groups seem to be separated. 
```{r PCA plot, echo=TRUE, message=FALSE, warning=FALSE}

plotPCA(vsd_dds, intgroup = c("condition","genotype"))
```




### Statistical Analysis 
```{r signifcant DEGs selection, cache=TRUE, include=FALSE}
sigresults_df <- result %>%
  as.data.frame() %>%
  dplyr::mutate(ID = gsub(pattern = "\\..*", replacement = "", x= rownames(result))) %>%
  dplyr::relocate(ID,baseMean) %>%
  arrange(desc(stat)) %>%
  subset(padj < 0.05) 
```

```{r Gene symbol ID selection, cache=TRUE, include=FALSE}
ensembl <- useEnsembl(biomart = "genes", dataset = "hsapiens_gene_ensembl")

sigresult_gene_id <- getBM(attributes = c("ensembl_gene_id","hgnc_symbol"),
                           filters = "ensembl_gene_id",
                           values = sigresults_df$ID,
                           mart = ensembl) %>%
                     dplyr::rename(ID = ensembl_gene_id, 
                                   Gene_name = hgnc_symbol)

sigresults_df <- left_join(sigresults_df,sigresult_gene_id, by= "ID") %>%
  dplyr::relocate(Gene_name,baseMean) %>%
  dplyr::select(ID,Gene_name,log2FoldChange,stat,pvalue,padj)

# Removes the rows with No gene names, not sure which we would prefer for the data table. If I had to guess I would assume we would want a table with Gene names so the results are more straightforward and we at least know each ensemble ID is unique and doesn't have a chance of overlapping with another gene name.    
sigresults_df_filtered <- sigresults_df %>%
  drop_na(Gene_name) %>%
  dplyr::filter(Gene_name != "")
  
```

### Significant DEGs table

This table list the Genes that have a adjusted p-value lower than 0.05, and sorted in descending order by the stat column. For this analysis we ended up with 493 significant DEGs. 
```{r sig DEGs datatable, echo=TRUE, message=FALSE, warning=FALSE}
datatable(sigresults_df_filtered, class = 'cell-border stripe' , 
          caption = "Table 2: Patient vs Control Condition Significant Genes")
```



### Enhanced Volcano plot

This volcano plot shows all of the significant DEGs, there is 5 clear groups of genes that seem to be split up mostly by their log2 Fold change, in the middle from (-5,5) contains the majority of DEGs, then is a little farther out is (-10) and (10), and at both ends we have the final 2 groups, (-20) and (20) which could show the genes that are the most overexpressed and underexpressed. The Fold Change cut off was adjusted to just leave the over- groups labeled since they might have the most interesting data to look into. 
```{r enhanced Volcano plot, echo=TRUE, message=FALSE, warning=FALSE}
EnhancedVolcano(sigresults_df_filtered, 
                lab = sigresults_df_filtered$Gene_name,
                x = 'log2FoldChange',
                y = 'pvalue',
                title ="volcano Slot of Signifcant DEGs",
                xlim = c(-28,28),
                ylim = c(0,17),
                pCutoff = 0.05,
                FCcutoff = 22,
                pointSize = 1.5,
                labSize = 2 )
```




### Heatmap of the top 10 overexpressed and top 10 underexpressed DEGs

```{r heatmap data input, cache=TRUE, include=FALSE}

topover <- dplyr::slice_max(sigresults_df_filtered,n = 10, order_by = stat)
topunder <- dplyr::slice_min(sigresults_df_filtered,n = 10, order_by = stat)
exprdata <- rbind(topover,topunder)

normalized_dds_counts <- counts(dds, normalized=TRUE)
row.names(normalized_dds_counts) <- str_replace(fullensemblID,
                          pattern = "\\..*", 
                          replacement = "")

sig_norm_dds_counts <- normalized_dds_counts[exprdata$ID,]
row.names(sig_norm_dds_counts) <- (exprdata$Gene_name)



heatmeta <- as.data.frame(metadata) %>%
  dplyr::select(conditions,genotype)


colsRdBu <- brewer.pal(11, "RdBu")
  
palette <- colorRampPalette(colsRdBu)

```



```{r heatmap plot, echo=TRUE, message=FALSE, warning=FALSE}
heatmap <- pheatmap(sig_norm_dds_counts,
                    color = palette(20),
                    cluster_rows = TRUE,
                    show_rownames = TRUE,
                    annotation = dplyr::select(heatmeta,conditions),
                    scale = "row")
```

```{r Clusterprofiler set up, message=FALSE, warning=FALSE, cache=TRUE, include=FALSE}

gene_list <- result_dataframe$stat %>%
  setNames(object = ,fullensemblID) %>%
  na.omit() %>%
  sort(decreasing = TRUE)

```

```{r Gene mapped, message=FALSE, warning=FALSE, cache=TRUE, include=FALSE}
gse <- gseGO(geneList = gene_list , ont = "ALL",
             keyType = "ENSEMBL",
             pvalueCutoff = 0.05, verbose = TRUE,
             OrgDb= "org.Hs.eg.db", pAdjustMethod = "holm")
```

### GSEA2 Plot for Top 5 Overexpressed and Underexpressed DEGs 


Here we have 2 GESA plots, one for the overexpressed pathways, and one for the underexpressed pathways. For this graph we have the top 5 overexpressed pathways which mean in neurons from patients who are at risk 
```{r GSEA2 Plot overexpressed, echo=TRUE, message=FALSE, warning=FALSE}
overGSEAplot <- gse %>%
  arrange(rank) %>%
  gseaplot2(geneSetID = 1:5, title = "Top 5 Overexpressed Pathways")
overGSEAplot
```

```{r GSEA2 Plot underexpressed, echo=TRUE, message=FALSE, warning=FALSE}
underGSEAplot <- gse %>%
  arrange(desc(rank)) %>%
  gseaplot2(geneSetID = 1:5, title = "Top 5 Underexpressed Pathways")
underGSEAplot
```

Discussion: 

Conclusion: 

